databricks/koalas/indexes.py

```diff
@@ -1884,6 +1884,30 @@ def union(self, other, sort=None):
 
         return DataFrame(internal).index
 
+    def holds_integer(self):
+        """
+        Whether the type is an integer type.
+
+        Examples
+        --------
+        >>> kidx = ks.Index([1, 2, 3, 4])
+        >>> kidx.holds_integer()
+        True
+
+        Returns False for string type.
+
+        >>> kidx = ks.Index(["A", "B", "C", "D"])
+        >>> kidx.holds_integer()
+        False
+
+        Returns False for float type.
+
+        >>> kidx = ks.Index([1.1, 2.2, 3.3, 4.4])
+        >>> kidx.holds_integer()
+        False
+        """
+        return isinstance(self.spark.data_type, LongType)
```

Comment:
Does it work if it contains nulls?

Reply:
Hmm.. good point.

If Index contains null values, the type of index will be `float64` in Koalas, so the result is gonna be different with pandas.

```python
>>> ks.Index([1, 2, None])
Float64Index([1.0, 2.0, nan], dtype='float64')

>>> pd.Index([1, 2, None])
Index([1, 2, None], dtype='object')
```

I'll add the different behaviour to the Notes since confirming whether the type of values except null are all integer might be expensive.

Reply:
I just added Notes !